components/tabs/src/sync/record.rs

```diff
@@ -27,7 +29,379 @@ pub struct TabsRecord {
 impl TabsRecord {
     #[inline]
     pub fn from_payload(payload: sync15::Payload) -> Result<Self> {
-        let record: TabsRecord = payload.into_record()?;
-        Ok(record)
+        // Note: We are hand-parsing the tabs payload so that we can provide support for
+        // older clients that may have a stringified float or integer `last_used` value.
+        let client_name: String =
```

Comment:
Stylistically, it feels like we shouldn't need the explicit `: String` here, unless I'm missing something? (Put another way, the presence of `: String` here made me wonder if I was missing something non-obvious about the behaviour of the called function that made the explicit type necessary, but I couldn't see anything)